File: examples/basics/cpp/basic-args.cpp
#include <iostream>

using namespace std;

int main(int argc, char *argv[]) {
  if (argc < 3)
    cout << "Error: missing argument firstname and legs number";
  else
    cout << "The dog is " << argv[1] << " and has " << argv[2] << " legs\n";
}

File: examples/basics/java/BasicArgs.java
public class BasicArgs {
  public static void main(String[] args) {
    if (args.length < 2)
      System.out.println("Error: missing argument firstname and legs number");
    else
      System.out.println("The dog is " + args[0] + " and has " + args[1] + " legs\n");
  }
}
